testing/asserts_test.ts

```diff
@@ -601,6 +603,8 @@ Deno.test("testingAssertObjectMatching", function (): void {
     assertObjectMatch({ test: { a: 1 } }, { test: new A(1) });
     assertObjectMatch({ test: new A(1) }, { test: new A(1) });
   }
+  // subset with TypedArray instance inside
+  assertObjectMatch(m, { bar: new Uint8Array([1, 2]) });
```

Comment:
The fix looks good to me, but this test case seems passing even without the change.

If I changed the assertion to `assertObjectMatch(m, m)`, then the issue described in #1842 reproduces as a test case. So maybe let's add that assertion here?